container-search/src/main/java/com/yahoo/prelude/query/NearestNeighborItem.java

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+package com.yahoo.prelude.query;
+
+import com.google.common.annotations.Beta;
+import com.yahoo.compress.IntegerCompressor;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Represent a query item matching the K nearest neighbors in a multi-dimensional vector space.
+ * The query point vector is referenced by the name of a tensor rank feature passed in the query,
+ * so specifying "myvector" as the name means the query must set "ranking.features.query(myvector)",
+ * which must be configured with the correct tensor type in the active query profile.
+ * The field name (AKA the index name) given must be an attribute, with the exact same tensor type.
+ *
+ * @author arnej
+ */
+@Beta
+public class NearestNeighborItem extends SimpleTaggableItem {
+
+    private int targetNumber = 0;
+    private String field;
+    private String property;
+
+    /** @return the K number of hits to produce */
+    public int getTargetNumHits() { return targetNumber; }
```

Comment:
Not: Constructor should go above accessors.

Reply:
Consider renaming member variable to targetNumHits as well.

Reply:
done